frontend/src/components/CuisineFilter.tsx

```diff
@@ -0,0 +1,94 @@
+import { cuisineList } from '@/config/restaurant-options-config';
+import { Label } from './ui/label';
+import { Check, ChevronDown, ChevronUp } from 'lucide-react';
+import { ChangeEvent } from 'react';
+import { Button } from './ui/button';
+
+type Props = {
+  onChange(cuisines: string[]): void;
+  selectedCuisines: string[];
+  isExpanded: boolean;
+  onExpandClick(): void;
+};
+
+const CuisineFilter = ({
+  onChange,
+  selectedCuisines,
+  isExpanded,
+  onExpandClick,
+}: Props) => {
+  const handleCuisinesReset = () => {
+    onChange([]);
+  };
+
+  const handleCuisinesChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const clickedCuisine = event.target.value;
+    const isChecked = event.target.checked;
+    const newCuisinesLint = isChecked
+      ? [...selectedCuisines, clickedCuisine]
+      : selectedCuisines.filter(
+          (cuisine: string) => cuisine !== clickedCuisine
+        );
+
+    onChange(newCuisinesLint);
+  };
+
+  return (
+    <>
+      <div className="flex justify-between items-center px-2">
+        <div className="text-md font-semibold mb-2">Filter by Cuisine</div>
+        <div
+          onClick={handleCuisinesReset}
+          className="text-md font-semibold mb-2 underline cursor-pointer text-blue-500"
+        >
+          Reset Filters
+        </div>
+      </div>
+
+      <div className="space-y-2 flex flex-col">
+        {cuisineList
+          .slice(0, isExpanded ? cuisineList.length : 7)
+          .map((cuisine) => {
+            const isSelected = selectedCuisines.includes(cuisine);
+            return (
+              <div className="flex">
+                <input
+                  id={`cuisine_${cuisine}`}
+                  type="checkbox"
+                  className="hidden"
+                  value={cuisine}
+                  checked={isSelected}
+                  onChange={handleCuisinesChange}
+                />
+                <Label
+                  htmlFor={`cuisine_${cuisine}`}
+                  className={`flex flex-1 item-center cursor-pointer text-sm rounded-full px-4 py-2 font-semibold ${
+                    isSelected
+                      ? 'border border-green-600 text-green-600'
+                      : 'border border-slate-300'
+                  }`}
+                >
+                  {isSelected && <Check size={20} strokeWidth={3} />}
+                  {cuisine}
+                </Label>
+              </div>
+            );
+          })}
+
+        <Button onClick={onExpandClick} variant="link" className="mt-4 flex-1">
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/44#issuecomment-2410860532>#3a7709</a><span> - 10/14/2024, 11:08 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The expand/collapse button lacks proper keyboard accessibility attributes, which may make it difficult for keyboard-only users to navigate and use the component effectively.
</div>
<div id="fix">
<b>Fix:</b> Add proper ARIA attributes to the expand/collapse button to improve its accessibility. Use 'aria-expanded' to indicate the current state and 'aria-controls' to associate it with the content it controls.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -78,6 +78,9 @@
 -        <Button onClick={onExpandClick} variant="link" className="mt-4 flex-1">
 +        <Button
 +          onClick={onExpandClick}
 +          variant="link"
 +          className="mt-4 flex-1"
 +          aria-expanded={isExpanded}
 +          aria-controls="cuisine-list"
 +        >
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged